onnxscript/ir/_core.py

```diff
@@ -1187,18 +1189,18 @@ def __init__(
         self._version: int | None = version
         self._metadata: _metadata.MetadataStore | None = None
         self._metadata_props: dict[str, str] | None = metadata_props
-        self._graph: Graph | Function | None = graph
+        # _graph is set by graph.append
+        self._graph: Graph | None = None
+        # Add the node to the graph if graph is specified
+        if graph is not None:
+            graph.append(self)
         self.doc_string = doc_string
 
         # Add the node as a use of the inputs
         for i, input_value in enumerate(self._inputs):
             if input_value is not None:
                 input_value._add_usage(self, i)  # pylint: disable=protected-access
 
-        # Add the node to the graph if graph is specified
-        if self._graph is not None:
-            self._graph.append(self)
```

Comment:
I don't understand how moving these up make any difference?

Reply:
I moved it closer to were graph is created so it is more readable.